spec/controllers/profile_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+describe ProfileController do
+  describe 'GET #index' do
+    context 'when user is not signed in' do
+      before { get :index, params: { 'uid': 'user-id' } }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context 'when user is logged in' do
```

Comment:
We want to test when a user is logged in and looking at their own profile and when a user is logged in a looking at someone else's profile